pkg/util/lock.go

```diff
@@ -108,7 +126,14 @@ func (l *PgAdvisoryLock) TryLock() (bool, error) {
 		return false, err
 	}
 	if l.obtained {
-		log.Debug("msg", fmt.Sprintf("Lock obtained for group id %d", l.groupLockId))
+		log.Debug("msg", fmt.Sprintf("Lock obtained for group id %d", l.groupLockID))
+		l.lockTimer = time.AfterFunc(l.lockTimeout, func() {
```

Comment:
I'd recommend not creating another timer, but instead relying on existing `ScheduledElector`. `ScheduledElector` is responsible for running leader check on certain interval . I guess we would just need to change `IsLeader` implementation of PG advisory lock to actually validate the connection and try re-obtaining the lock. In this case election interval would need to be configurable and default to something more reasonable. (eg. 5 seconds).